### PR TITLE
implement `EXPLAIN` and `EXPLAIN PLAN`

### PR DESCRIPTION
**Pull request by jycor**
Moving our current implementation of `EXPLAIN` to `EXPLAIN PLAN`, and replace `EXPLAIN` with a dummy implementation of MySQL's `EXPLAIN`

Looks like this now:
```sql
tmp2/main> explain select * from t;
+----+-------------+-------+------------+------+---------------+------+---------+------+------+----------+-------+
| id | select_type | table | partitions | type | possible_keys | key  | key_len | ref  | rows | filtered | Extra |
+----+-------------+-------+------------+------+---------------+------+---------+------+------+----------+-------+
| 1  | SELECT      | NULL  | NULL       | NULL | NULL          | NULL | NULL    | NULL | NULL | NULL     |       |
+----+-------------+-------+------------+------+---------------+------+---------+------+------+----------+-------+
1 row in set (0.00 sec)

tmp2/main> explain plan select * from t;
+------------------+
| plan             |
+------------------+
| Table            |
|  ├─ name: t      |
|  └─ columns: [i] |
+------------------+
3 rows in set (0.00 sec)

```

related: https://github.com/dolthub/dolt/issues/8592

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -43,4 +43,4 @@ require (
 	gopkg.in/check.v1 v1.0.0-20200902074654-038fdea0a05b // indirect
 )
 
-go 1.22.2
+go 1.23.3
```
Comment (max-hoffman) [inline go.mod:46]: do you wanna do this in this PR?

**Review by max-hoffman — APPROVED** [review]
lgtm, seems like it might be a bit disruptive

File: go.mod (hunk 1 above)
Comment (jycor) [reply go.mod:46]: might as well

File: sql/rowexec/show.go [hunk 2]
```diff
@@ -74,14 +74,31 @@ func (b *BaseBuilder) buildDescribeQuery(ctx *sql.Context, n *plan.DescribeQuery
...
+				rows = append(rows, sql.NewRow(l))
+			}
 		}
+		return sql.RowsToRowIter(rows...), nil
 	}
-	return sql.RowsToRowIter(rows...), nil
+
+	dummyRow := sql.Row{
+		1,        // id
+		"SELECT", // select_type
```
Comment (max-hoffman) [inline sql/rowexec/show.go:89]: maybe include a warning as long as the output isn't useful

File: go.mod (hunk 1 above)
Comment (jycor) [reply go.mod:46]: oh wait, this is from vitess having a newer version of go...

**Pull request merged**